test/test_serialization.py

```diff
@@ -1057,6 +1057,19 @@ def _save_load_check(t):
         with self.assertRaisesRegex(RuntimeError, 'ZeroTensor is not serializable'):
             _save_load_check(t)
 
+    @parametrize('pickle_protocol', (2, 3, 4, 5))
+    @parametrize('weights_only', (False, True))
+    def test_serialization_pickle_protocol_version(self, pickle_protocol, weights_only):
```

Comment:
nit: Hey @jerrychenhf, thanks for your patience on this @malfet will review this from a security perspective, but just wanted to add some context here

My understanding from our discussion on the linked issue is that the specific case that this targets for weights_only is tensors that go through numpy when serialization that are > 4GB, which will require pickle protocol > 4

https://github.com/pytorch/pytorch/blob/abdd569e412c4af5957b10ff8039511f8175986e/torch/_tensor.py#L262-L277

In that case it might be good to add a test re this, that illustrates this. Fwiw the weights_only unpickler doesn't run with numpy serialized tensors right now, but it should with this change https://github.com/pytorch/pytorch/pull/124763

My understanding is that [this](https://github.com/pytorch/pytorch/blob/main/test/test_cpp_extensions_open_device_registration.py?fbclid=IwAR0yeivTmqoBQEI6Zz98zRMVh_WBo09w-MgQxy-TAI0tbtuq4n_W7AWxC4o) would be a good file to add such a test, however the test might not be active in CI right now due to #100152 (which should be fixed soon)

Reply:
@mikaylagawarecki Thank you for the information.
Yes, The current weights_only unpickler cannot load numpy serialized tensors. A few related classes are not in the allowed list.
I will try work out a test case as you suggest. For the new test case to pass (assume it is active after https://github.com/pytorch/pytorch/issues/100152 issue is fixed), we would need https://github.com/pytorch/pytorch/pull/124763 to be merged first.